File: 450/Ratndeep/Array/middle of three.py

class Solution:
    def middle(self,A,B,C):
        #code here
        if A<B:
            return B if B<C else max(A,C)
        return A if A<C else max(B,C)


s = Solution()
a,b,c = map(int,input().split())
print(s.middle(a,b,c))

File: 450/Utkarsh/array/middleOfThree.cpp
//https://practice.geeksforgeeks.org/problems/middle-of-three2926/1

#include<bits/stdc++.h>
using namespace std;

// NO comparisons required at all
class Solution{
  public:
    int middle(int A, int B, int C){
        int maxi  = max(A,B)     , mini  = min(A,B);
        int maxii = max(maxi,C)  , minii = min(mini,C);
        return A+B+C-maxii-minii;
    }
};